node/lib/cmd/add.js

```diff
@@ -92,14 +92,40 @@ and avoid accidental changes to the meta-repo.`,
  * @param {String[]} args.paths
  */
 exports.executeableSubcommand = co.wrap(function *(args) {
-    const Add     = require("../util/add");
-    const GitUtil = require("../util/git_util");
+    const colors = require("colors");
+
+    const Add        = require("../util/add");
+    const GitUtil    = require("../util/git_util");
+    const StatusUtil = require("../util/status_util");
 
     const repo    = yield GitUtil.getCurrentRepo();
     const workdir = repo.workdir();
     const cwd     = process.cwd();
 
-    const paths = args.paths.map(filename => {
+    let userPaths = args.paths;
+    if (userPaths.length === 0) {
+        if (args.update) {
+            const repoStatus = yield StatusUtil.getRepoStatus(repo, {
+                cwd: cwd,
+                paths: args.path,
+                untrackedFilesOption: args.untrackedFilesOption
+            });
+
+            const fileStatuses = exports.accumulateStatus(repoStatus);
+            const workdirChanges = fileStatuses.workdir;
+            userPaths = workdirChanges.map(workdirChange => {
+                return workdirChange.path;
+            });
+        }
+        else {
+            const text = "Nothing specified, nothing added.\n" +
+                         `${colors.yellow("hint: Maybe you wanted to say ")}` +
```

Comment:
Is the use of yellow here for compatibility with Git?  I find yellow on a white background very hard to read.